src/test/java/com/arpnetworking/metrics/impl/BaseFileSinkTest.java

```diff
@@ -129,7 +137,10 @@ public void testBuilderWithNull() throws IOException {
         Assert.assertEquals(24, rollingPolicy.getMaxHistory());
         Assert.assertTrue(encoder.isImmediateFlush());
         Assert.assertEquals(expectedPath + "query.log", rollingAppender.getFile());
-        Assert.assertEquals(expectedPath + "query.%d{yyyy-MM-dd-HH}.log.gz", rollingPolicy.getFileNamePattern());
+        Assert.assertEquals(expectedPath + "query.%d{yyyy-MM-dd-HH}.%i.log.gz", rollingPolicy.getFileNamePattern());
+
+        // Not currently implemented by logback
+//        Assert.assertEquals("100MB", rollingPolicy.getMaxFileSize());
```

Comment:
Given the issues with size based rolling I think it would be prudent to include an integration test. For example, set a very low threshold so each even is written to its own log file. You can have a few simple tests and then a more complex test where you exceed the maximum number of files based on size only and that the oldest file(s) is deleted.